src/model_ode.cpp

```diff
@@ -316,9 +320,10 @@ void Model_ODE::fxBdot(realtype t, N_Vector x, N_Vector xB, N_Vector xBdot) {
 void Model_ODE::fqBdot(realtype t, N_Vector x, N_Vector xB, N_Vector qBdot) {
     N_VConst(0.0, qBdot);
     fdxdotdp(t, x);
+    /* CHANGE_TO_SPARSE --> This loop needs to be changed... */
     for (int ip = 0; ip < nplist(); ip++) {
         for (int ix = 0; ix < nxtrue_solver; ix++)
-            NV_Ith_S(qBdot, ip * nJ) -= NV_Ith_S(xB, ix) * dxdotdp.at(ix, ip);
+            NV_Ith_S(qBdot, ip * nJ) -= NV_Ith_S(xB, ix) * dxdotdp.data(ip)[ix];
```

Comment:
Do we want to modify `at()` to default to unchecked access? This might be somewhat unexpected. Maybe just add a separate function for unchecked access, or better save a reference to `auto cur_dxdotdp = dxdotdp[ip];`and in the inner loop use `cur_dxdotdp[ix]`.

Reply:
I agree here with Daniel that we should keep checked access. It's only costly in this specific case, but doesn't substantially harm in others... Anyway, this code part will soon not use `.at()` at all anymore...